### PR TITLE
Start documenting enums

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by alex** [comment]
@dreid the first of the feature descriptions, as we idscussed.

File: docs/features/enums.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+The Bagel compiler will make sure that you handle all possible cases, so if you
+add a new value to ``MeasuringSystem`` without adding handling to
+``distance_unit`` you will get a compilation error, tell you about the
+unhandled case.
+
+Unlike many programming languages, Bagel's enums can contain other values::
+
+    enum class Distance
+        Inches(UInt)
+        Feet(UInt)
```
Comment (dreid) [inline docs/features/enums.rst:30]: The repetition of UInt here would seem to imply that the following would be valid:

```
enum class Distance:
    Inches(Uint)
    Feet(Text)
```

Also your example is missing a ':'

Comment (alex) [inline docs/features/enums.rst:30]: The code you wrote is valid. Did you mean to say /wouldn't/ be valid?

Comment (dreid) [inline docs/features/enums.rst:30]: If that is valid (and I don't see why it shouldn't be) you might explain that it is valid and give an example of when you'd use it?

**Comment by alex** [comment]
Do you have a suggestion for a use case (about to board an airplane :P)?

File: docs/features/enums.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Then we can declare functions which take one of these values, and use
+:term:`pattern matching` to handle them::
+
+    def distance_unit(system: MeasuringSystem) -> Text:
+        match system:
+            as Imperial:
+                return "Feet"
+            as Metric:
+                return "Meters"
```
Comment (dreid) [inline docs/features/enums.rst:19]: Maybe you'll explain it in the pattern matching documentation, but will these have default branches?

If not maybe the previously mentioned feature of values containing different types could be used to specify fallbacks so you can have an interface "extending" enums used by a library.

```
enum class MeasuringSystem:
    Imperial
    Metric
    Unknown(MeasuringSystemDescriptor)
```

**Pull request merged**